lib/APPReinforcementLearning/src/App.cpp

```diff
@@ -106,29 +110,76 @@ void App::loop()
          */
         Odometry::getInstance().process();
 
-#ifdef DEBUG_ODOMETRY
+        m_controlInterval.restart();
+    }
+
+    if ((true == m_statusTimer.isTimeout()) && (true == m_smpServer.isSynced()) &&
+        (&DrivingState::getInstance() == m_systemStateMachine.getState()))
+    {
+        Status payload = {SMPChannelPayload::Status::NOT_DONE};
+
+        if (DrivingState::getInstance().isAbortRequired() == true)
         {
-            Odometry&    odo         = Odometry::getInstance();
-            int32_t      posX        = 0;
-            int32_t      posY        = 0;
-            int32_t      orientation = odo.getOrientation();
-            const size_t BUFFER_SIZE = 128U;
-            char         buffer[BUFFER_SIZE];
+            payload = {SMPChannelPayload::Status::DONE};
+        }
 
-            odo.getPosition(posX, posY);
+        /* Ignoring return value, as error handling is not available. */
+        (void)m_smpServer.sendData(m_serialMuxProtChannelIdStatus, &payload, sizeof(payload));
 
-            snprintf(buffer, BUFFER_SIZE, "ODO,%d,%d,%d", posX, posY, orientation);
+        m_statusTimer.restart();
+    }
 
-            Board::getInstance().getSender().send(buffer);
-        }
-#endif /* DEBUG_ODOMETRY */
+    /* Send periodically line sensor data. */
+    if (true == m_sendLineSensorsDataInterval.isTimeout() &&
+        (&DrivingState::getInstance() == m_systemStateMachine.getState()))
+    {
+        sendLineSensorsData();
 
-        m_controlInterval.restart();
+        m_sendLineSensorsDataInterval.restart();
+    }
+
+    /* Send Mode selected to The Supervisor. */
+    if (&ReadyState::getInstance() == m_systemStateMachine.getState() && (!m_modeSelectionSent))
```

Comment:
Please use explicit comparison.
```suggestion
    if (&ReadyState::getInstance() == m_systemStateMachine.getState() && (false == m_modeSelectionSent))
```